base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -973,6 +1054,29 @@ impl BlockchainBackend for LMDBDatabase {
         })
     }
 
+    fn fetch_tip_header(&self) -> Result<ChainHeader, ChainStorageError> {
```

Comment:
Note this change. Previously would return the last header, now returns the header at tip according to metadata